File: questions/codeforce/Codeforces Round #598 (Div. 3)/A.py
def solve(a, b, n, S):
    l, r = 0, a
    while l <= r:
        mid = (l+r)//2;
        if mid*n == S or ((mid*n)+b >= S and mid*n < S):
            print('YES')
            return 1
        elif mid*n > S:
            r = mid - 1
        else:
            l = mid + 1
    print('NO')
    return 0
    
for i in range(int(input())):
    solve(*list(map(lambda i: int(i), input().split())))

File: questions/codeforce/Codeforces Round #598 (Div. 3)/A.cpp

#include <bits/stdc++.h> 

int q, a, b, n, S;
double res;

int solve(){
	int Scopy = S;
	double t = S/n;
	while (abs(S/n) != S/n){
		S-=1;
		t = S/n;
		if (S<=0){
			std::cout<<"NO"<<"\n";
			return 0;
		}
	}
	if (t<=a and (n*t)+b >= Scopy){
		std::cout<<"YES"<<"\n";
		return 0;
	}else{
		std::cout<<"NO"<<"\n";
		return 0;
	}

}

int main()  
{ 
	std::cin>>q;
	for(int i = 0; i < q; i++){
		std::cin>>a>>b>>n>>S;
		//std::cout<<a<<" "<<b<<" "<<n<<" "<<S<<" ";
		solve();
	}
    return 0; 
} 